pipelines/rbsa/tasks/find_sensitivities.py

```diff
@@ -0,0 +1,212 @@
+        self.df = self.df.set_index(['time'])
+        self.df.index = pd.to_datetime(self.df.index)
+
+        if self.df.columns[0] == 'Unnamed: 0':
+            self.df = self.df.drop('Unnamed: 0', axis=1)
+
+        loadshapes = pd.DataFrame(columns=self.df.columns)
+
+        zipcodes = self.df.zipcode.unique()
+
+        loadshapes = pd.DataFrame(columns=list(self.df.columns))
+
+        for zipcode in zipcodes:
+
+            weather_file = f'noaa/{str(zipcode)}.csv'
+            weather = data_map[weather_file]
+            weather = weather.set_index(['DATE'])
+            weather.index = pd.to_datetime(weather.index)
+
+            zipcode_df = self.df.loc[self.df.zipcode == zipcode]
+
+            if weather.shape[0] > zipcode_df.shape[0]:
+                weather = weather.loc[weather.index.isin(zipcode_df.index)]
+
+            self.enduse_cols = list(zipcode_df.columns)
+            self.enduse_cols.remove('zipcode')
+
+            A = self.get_A(weather)
+
+            zipcode_loadshapes = pd.DataFrame(columns=self.enduse_cols)
+
+            for enduse in self.enduse_cols:
+                enduse_df = zipcode_df[enduse]
+
+                if enduse == 'Heating':
+                    A_heat = A[:,:-1]
+                    x = self.get_baseload(enduse_df, A_heat)
+                    x = np.append(x, [0])
+
+                elif enduse == 'Cooling':
+                    A_cool = np.delete(A, -2, 1)
+                    x = self.get_baseload(enduse_df, A_cool)
+                    cooling_sensitivity = x[-1]
+                    x = np.append(x[:-1], [0])
+                    x = np.append(x, [cooling_sensitivity])
+
+                else:
+                    A_base = A[:,:-2]
+                    x = self.get_baseload(enduse_df, A_base)
+                    x = np.append(x, [0])
+                    x = np.append(x, [0])
+
+                heat_sens = x[-2]
+                cool_sens = x[-1]
+
+                # adjust x by first value
+                x = np.append(x[0],x[1:48]+x[0])
+                x = np.append(x,[heat_sens, cool_sens])
+
+                zipcode_loadshapes[enduse] = x
+
+            zipcode_loadshapes.insert(loc=0, column='zipcode', value=zipcode)
+
+            # print(zipcode_loadshapes.head(4))
+
+
+            loadshapes = loadshapes.append(zipcode_loadshapes)
+
+        self.validate(loadshapes)
+        self.on_complete({self.output_artifact_loadshapes: loadshapes})
+
+    def get_baseload(self, df, A):
+        """ This function gets baseload for non weather sensitive enduses
+        """
+
+        At = A.transpose()
+        y = np.asarray(df).transpose()
+        M = np.matmul(np.linalg.inv(np.matmul(At,A)),At)
+        x = np.matmul(M,y)
+
+        return x
+
+    def get_nonsensitive_A(self, weather):
+        """constructs A matrix for non weather sensitive loads
+        """
+
+        A = np.zeros((len(weather.index),48),float)
+
+        ts = datetime.datetime(weather.index[0].year,1,1,0,0,0)
```

Comment:
Need a single whitespace after the commas

Reply:
fixed